tardis/gui.py

```diff
@@ -43,68 +105,739 @@ def __init__(self, parent=None):
+        
+        #Data structures
         self.model = None
         self.shell_info = {}
         self.line_info = []
-        self.setGeometry(20, 35, 1250, 500)
-        self.setWindowTitle('Shells Viewer')
-        self.tablemodel = SimpleTableModel([['Shell: '], ["Rad. temp", "Ws"]], (1, 0))
+
+        #Shells widget
+        self.shellWidget = self.makeShellWidget()
+        
+        #Spectrum widget
+        self.spectrumWidget = self.makeSpectrumWidget()
+
+        #Plot tab widget
+        self.plotTabWidget = QtGui.QTabWidget()
+        self.plotTabWidget.addTab(self.shellWidget,"&Shells")
+        self.plotTabWidget.addTab(self.spectrumWidget, "S&pectrum")
+
+        #Table widget
+        self.tablemodel = SimpleTableModel([['Shell: '], ["Rad. temp", "Ws"]],
+         (1, 0))
         self.tableview = QtGui.QTableView()
+        self.tableview.setMinimumWidth(200)
+        self.tableview.connect(self.tableview.verticalHeader(), 
+            QtCore.SIGNAL('sectionClicked(int)'), self.graph.highlight_shell)
+        self.tableview.connect(self.tableview.verticalHeader(), 
+            QtCore.SIGNAL('sectionDoubleClicked(int)'),
+            self.on_header_double_clicked)
+
+        #Label for text output
+        self.outputLabel = QtGui.QLabel()
+        self.outputLabel.setFrameStyle(QtGui.QFrame.StyledPanel | 
+            QtGui.QFrame.Sunken)
+        self.outputLabel.setStyleSheet("QLabel{background-color:white;}")
+
+        #Group boxes
+        graphsBox = QtGui.QGroupBox("Visualized results")
+        textsBox = QtGui.QGroupBox("Model parameters")
+        tableBox = QtGui.QGroupBox("Tabulated results")
+
+        #For textbox
+        textlayout = QtGui.QHBoxLayout()
+        textlayout.addWidget(self.outputLabel)
+
+        tableslayout = QtGui.QVBoxLayout()
+        tableslayout.addWidget(self.tableview)
+        tableBox.setLayout(tableslayout)
+
+        visualayout = QtGui.QVBoxLayout()
+        visualayout.addWidget(self.plotTabWidget)
+        graphsBox.setLayout(visualayout)
+
+        self.layout = QtGui.QHBoxLayout()
+        self.layout.addWidget(graphsBox)
+        textntablelayout = QtGui.QVBoxLayout()
+        textsBox.setLayout(textlayout)
+        textntablelayout.addWidget(textsBox)
+        textntablelayout.addWidget(tableBox)
+
+        self.layout.addLayout(textntablelayout)                
+        self.setLayout(self.layout)
+
+    def fillOutputLabel(self):
+        labeltext = 'Iterations requested: {} <br/> Iterations executed:  {}<br/>\
+                     Model converged     : {} <br/> Simulation Time    :  {} s <br/>\
+                     Inner Temperature   : {} K <br/> Number of packets  :  {}<br/>\
+                     Inner Luminosity    : {}'\
+                     .format(self.model.iterations_max_requested, 
+                        self.model.iterations_executed,
+                        '<font color="green"><b>True</b></font>' if 
+                        self.model.converged else 
```

Comment:
very close. you should use string formatting. so do: `"<font color='green'><b>{0}</b></font>".format(self.model.converged)